test/prob/neg_binomial/neg_binomial_test.hpp

```diff
@@ -75,8 +69,6 @@ class AgradDistributionsNegBinomial : public AgradDistributionTest {
     using stan::math::multiply_log;
     using std::log;
 
-    if (alpha > 1e10)
-      return -lgamma(n + 1.0) + multiply_log(n, alpha / beta) - alpha / beta;
```

Comment:
I deleted the test values above because they were causing tests to fail with `as_scalars_vs_as_vector` (the new tests).

I'm pretty sure the reason they were failing is because:

1. The test is testing crazy large lpmf values that are at the limit of the numerics.

2. The new `as_scalars_vs_as_vector` stuff mixes and matches parameters from different sets of parameters.

For instance, if parameter 1 is a vector it will get all possible values of parameter 1, but if parameter 2 is a scalar, it will only get one value (and so different parameter 1s and parameter 2s get evaluated together). This is harder numerically than the hard numerical example, and things break.

----

I removed the branching logic in the tests cause that was only for very large parameters. This sort of branching used to exist in the regular implementations but @martinmodrak found a way to fix the numerics and avoid this . I think I did the `neg_binomial` changes for this: https://github.com/stan-dev/math/commit/56988042310b7d28eaddbf2c4fde2799884a3339#diff-43a81d5aa08039bf4bdfa75818ff97b8

I can add gradient checks at those large parameter values easily enough. 

----

I can also roll back `as_scalars_vs_as_vectors` I think. The changes I made to how the gradients are compared is what made the parameter shuffling happen that causes this.

Reply:
And if I roll back the checks then this test can stay.

Reply:
Nevermind I think this stays like this.

If we're testing mixes of scalar/vector types, we're going to end up in a situation like this where two sets of parameters just shuffled together and we get weird unexpected tests.

The way to get around this is only test repeated things, which is what was already being tested in the test framework, but was not sufficient to catch the original bugs.

Reply:
If this used to be tested on two parameter sets, I would like it to stay that way. Can you not find a second parameter set that mixes with the first one without numerical issues?

Reply:
I switched to expect_near_rel and was able to add this back.